cmd/restic/lock_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"github.com/restic/restic/internal/repository"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/restic/restic/internal/test"
+)
+
+func TestLockWaitTimeout(t *testing.T) {
+	repo, cleanup := repository.TestRepository(t)
+	defer cleanup()
+
+	elock, err := lockRepoExclusive(repo.(*repository.Repository))
+	test.OK(t, err)
+
+	wait := 2 * time.Second
+	oldWaitLock := globalOptions.WaitLock
+	globalOptions.WaitLock = wait
+	start := time.Now()
+	lock, err := lockRepo(repo.(*repository.Repository))
+	duration := time.Since(start)
+	globalOptions.WaitLock = oldWaitLock
+	test.Assert(t, err != nil,
+		"create normal lock with exclusively locked repo didn't return an error")
+	test.Assert(t, strings.Contains(err.Error(), "repository is already locked exclusively"),
```

Comment:
I'm not happy with this substring check, but https://github.com/restic/restic/blob/870e7583a1128c9a6b20932178620ee2bafe0ef4/cmd/restic/lock.go#L38-L40 prevents me from using IsAlreadyLocked(err) here.